component/portal/src/main/java/org/exoplatform/portal/mop/rest/NavigationRest.java

```diff
@@ -265,7 +273,17 @@ private List<ResultUserNode> convertNodes(Collection<UserNode> nodes) {
         continue;
       }
       ResultUserNode resultNode = new ResultUserNode(userNode);
-      resultNode.setChildren(convertNodes(userNode.getChildren()));
+      if (expandPageDetails && userNode.getPageRef() != null) {
+        Page userNodePage = pageStorage.getPage(userNode.getPageRef());
```

Comment:
use  Page getPage(PageKey pageKey) of layoutService (service layer) instead of pageStorage (storage layer)